packages/offix-offline/src/utils/cacheHelper.ts

```diff
@@ -11,8 +11,14 @@ export const getObjectFromCache = (operation: Operation, id: string) => {
     const context = operation.getContext();
 
     if (context.cache && context.cache.data) {
-      const cacheData = context.cache.data.data;
       const idKey = context.getCacheKey({ __typename: context.returnType, id });
+      if (context.cache.optimisticData && context.cache.optimisticData.parent) {
+        const optimisticData = context.cache.optimisticData.parent.data;
+        if (idKey && optimisticData[idKey]) {
+          return Object.assign({}, optimisticData[idKey]);
```

Comment:
any reason why we create new object? This is going to ignore nested elements anyway.